### PR TITLE
fix(app): prevent LPC from failing on launch with probe on Flex robots

### PR DESCRIPTION
**Pull request by b-cooper**
# Overview

Repair the faulty separation of probe-based vs. tip-based LPC to cleanly fall on Flex vs. OT-2
respectively

Closes [RQA-1831](https://opentrons.atlassian.net/browse/RQA-1831), Closes [RQA-1830](https://opentrons.atlassian.net/browse/RQA-1830)

# Review requests

- Run metal probe-based LPC on a Flex

# Risk assessment
med

[RQA-1831]: https://opentrons.atlassian.net/browse/RQA-1831?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[RQA-1830]: https://opentrons.atlassian.net/browse/RQA-1830?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/src/organisms/LabwarePositionCheck/LabwarePositionCheckComponent.tsx [hunk 1]
```diff
@@ -250,10 +250,15 @@ export const LabwarePositionCheckComponent = (
...
-  const LPCSteps = getLabwarePositionCheckSteps(protocolData, robotType)
+  const LPCSteps = getLabwarePositionCheckSteps(
+    protocolData,
+    shouldUseMetalProbe
+  )
   const totalStepCount = LPCSteps.length - 1
   const currentStep = LPCSteps?.[currentStepIndex]
   if (currentStep == null) return null
+  console.log('currentStep', currentStep)
+  console.log('protocolData in top', protocolData)
```
Comment (brenthagen) [inline app/src/organisms/LabwarePositionCheck/LabwarePositionCheckComponent.tsx]: console.log

File: app/src/pages/OnDeviceDisplay/ProtocolSetup/index.tsx [hunk 2]
```diff
@@ -365,7 +366,8 @@ function PrepareToRun({
     protocolRecord?.data.files[0].name ??
     ''
   const mostRecentAnalysis = useMostRecentCompletedAnalysis(runId)
-  const { launchLPC, LPCWizard } = useLaunchLPC(runId, protocolName)
+  const robotType = useRobotType(robotName)
```
Comment (brenthagen) [inline app/src/pages/OnDeviceDisplay/ProtocolSetup/index.tsx:369]: is using this hook on ODD for future-proofing?

**Review by brenthagen — APPROVED** [review]
code LGTM !

**Review by skowalski08 — APPROVED** [review]
tested with Brian Cooper

**Pull request merged**